packages/eddsa-poseidon/src/eddsa-poseidon.ts

```diff
@@ -53,7 +53,7 @@ export function deriveSecretScalar(privateKey: Buffer | Uint8Array | string): bi
     hash = hash.slice(0, 32)
     hash = pruneBuffer(hash)
 
-    return scalar.shiftRight(leBufferToBigInt(hash), BigInt(3))
+    return scalar.shiftRight(leBufferToBigInt(hash), BigInt(3)) % subOrder
```

Comment:
can you confirm `subOrder` is `2736030358979909402780800718157159386076813972158567259200215660948447373041`?

I think so but still good to confirm from you.

Reply:
Just double-checked! It's right